### ISSUE TITLE
Passing a value to a not nullable parameter should update state

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#43383]
LDM is discussing details of the proposal, but we agreed on core scenario:

``` csharp
void M(string s) { }
...
string? s = null;
M(s); // warn only here
s.ToString(); // don't warn again
```

This means that `void M(string s)` would behave like `void M([NotNull] string s)` behaves today.

FYI @stephentoub 

----
The latest proposal is we can change the nullable state of a slot when either (1) we give a warning, or (2) you apply the `!` operator.

----
Update: LDM decision [below](https://github.com/dotnet/roslyn/issues/43383#issuecomment-638438735)

**Comment by stephentoub** [comment]
Thanks, @jcouv.

**Comment by jcouv** [comment]
From LDM today we're going to align behavior of `void M(string s)`, `void M([DisallowNull]string s)` and `void M([NotNull]string s)`. We'll change the state of the argument if it is not suppressed with `!`.

**Comment by RikkiGibson** [comment]
my interpretation of this is that while a not-nullable parameter updates the argument state, an oblivious parameter does not update the argument state

**Comment by jcouv** [comment]
> My interpretation of this is that while a not-nullable parameter updates the argument state, an oblivious parameter does not update the argument state.

Correct
